src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Binders/HeaderModelBinder.cs

```diff
@@ -45,86 +45,79 @@ public HeaderModelBinder(ILoggerFactory loggerFactory)
             _logger = loggerFactory.CreateLogger<HeaderModelBinder>();
         }
 
+        /// <summary>
+        /// Initializes a new instance of <see cref="HeaderModelBinder"/>.
+        /// </summary>
+        /// <param name="innerModelBinder">The <see cref="IModelBinder"/> which does the actual
+        /// binding of values.</param>
+        /// <param name="loggerFactory">The <see cref="ILoggerFactory"/>.</param>
+        public HeaderModelBinder(ILoggerFactory loggerFactory, IModelBinder innerModelBinder)
```

Comment:
**Question**: This HeaderModelBinder now delegates binding to the inner model binder. For users who upgrade to Preview2 and who use the HeaderModelBinderProvider as it is, then this constructor would be used and all is well, but if a user had been explicitly(the parameterless or the one taking in ILoggerFactory) using the HeaderModelBinder constructors, then the InnerModelBinder would be null breaking their application at runtime. We should have a backup (i.e use earlier design) in case the inner binder is null?

Reply:
Yep. That's the non-breaking change way to do this.

Reply:
I'd try to keep 1-2 tests that use the old code path and make the majority of them go down the new code path.